marlowe-dashboard-client/src/Play/State.purs

```diff
@@ -94,12 +100,15 @@ handleAction CloseCard = do
   for_ (init cards) \remainingCards ->
     assign _cards remainingCards
 
--- FIXME: probably get rid of this and tell ContractHome via MainFrame.handleQuery (getting the
---        currentSlot from the PAB)
-handleAction (SetCurrentSlot currentSlot) = do
-  toContractHome $ ContractHome.handleAction $ ContractHome.AdvanceTimedOutContracts currentSlot
-  modify_
-    $ set _currentSlot currentSlot
+handleAction (SetCurrentSlot slot) = do
+  toContractHome $ ContractHome.handleAction $ ContractHome.AdvanceTimedOutContracts slot
+  assign _currentSlot slot
+
+handleAction (CheckCurrentSlot slot) = do
+  currentSlot <- use _currentSlot
+  let
+    maxToleratedDifference = Slot $ fromInt 2
```

Comment:
I think up to 10 seconds we should be ok... 2 will probably cause a lot of false positives